src/transformStateWithClones.js

```diff
@@ -7,7 +9,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateVersions = [];
+
+  actions.reduce((currentState, action) => {
+    let nextState;
+
```

Comment:
Use the destruction for `action`
```suggestion
  actions.reduce((currentState, action) => {
    let nextState;
    const { type, ........ } = action; 
```